src/Donations/Endpoints/ListDonations.php

```diff
@@ -151,6 +152,16 @@ public function handleRequest(WP_REST_Request $request): WP_REST_Response
         );
     }
 
+    /**
+     * Initialize which list table to use.
+     *
+     * @unreleased
+     */
+    protected function init ()
+    {
+        $this->listTable = give(DonationsListTable::class);
```

Comment:
Instead of extracting this to a specific use `init()` method, I would prefer to use dependency injection from the container via the constructor. Currently, we are resolving the class from the container using the `give()` helper method, but moving this dependency to the `__construct()` as a parameter will still use the container, but can be  overloaded by an extending class.

```php
// givewp/src/Donations/Endpoints/ListDonations.php
class ListDonations extends Endpoint {
    public function __construct(DonationsListTable $listTable) {
        $this->listTable = $listTable;
    }
}
```


```php
// my-plugin/my-list-donations.php
class MyListDonations extends MyListDonations {
    public function __construct(MyDonationsListTable $myListTable) {
        $this->listTable = $myListTable;
    }
}
```

Reply:
Good morning,

Thank you for your quick response and your interest in my request.

I understood your remarks well. I had thought about this possibility of implementation.

There is a performance issue here. If another API route called, this ListTable will be useless instantiated. This is why I didn't want to instantiate the ListTable during construction.

The instantiation should still be triggered by `handleRequest`, in my opinion. What do you think of this proposal:

```php
// givewp/src/Donations/Endpoints/ListDonations.php
class ListDonations extends Endpoint {
    /**
     * The ListTable to which provide the data
     * @var string
     */
    protected const LIST_TABLE = DonationsListTable::class;

    // ...

    public function handleRequest(WP_REST_Request $request): WP_REST_Response {
        $this->listTable = Give(static::LIST_TABLE);
        $this->request = $request;
        // ...
    }
}
```

```php
// my-plugin/my-list-donations.php
class MyListDonations extends ListDonations {
    protected const LIST_TABLE = MyDonationsListTable::class;
}
```